ilmoituslomake/opening_times/utils.py

```diff
@@ -0,0 +1,208 @@
+    id="kaupunkialusta",
+    name_fi=None,
+    name_sv=None,
+    name_en=None,
+):
+    """
+    Update the origin of the given target.
+    Returns the response json
+    """
+    # Get the existing data
+    try:
+        response = requests.get(HAUKI_API_URL + str(hauki_id) + "/")
+    except requests.exceptions.HTTPError as errh:
+        print("Http Error:", errh)
+    except requests.exceptions.ConnectionError as errc:
+        print("Error Connecting:", errc)
+    except requests.exceptions.Timeout as errt:
+        print("Timeout Error:", errt)
+    except requests.exceptions.RequestException as err:
+        print("OOps: Something Else", err)
+
+    # Check if response is successful
+    if response.status_code != 200:
+        return response.json()
+
+    # If the notification is a draft add "draft-" -prefix.
+    if is_draft:
+        origin_id = "draft-" + str(origin_id)
+    # Add the new origin to existing origins.
+    origin = {
+        "data_source": {
+            "id": id,
+        },
+        "origin_id": origin_id,
+    }
+
+    # Add the new origin to existing ones.
+    origins = []
+    for item in response.json()["origins"]:
+        if item["data_source"]["id"] != "kaupunkialusta":
+            origins.append(item)
+
+    origins.append(origin)
+
+    update_params = {
+        "origins": origins,
+    }
+
+    # Partially update the resource
+    partially_update_hauki_resource(HAUKI_API_URL + str(hauki_id) + "/", update_params)
+
+
+def create_hauki_resource(
+    name, description, address, resource_type, origins, is_public, timezone
+):
+    # Authorization header.
+    authorization_headers = {"Authorization": "APIToken " + API_TOKEN}
+    # Params to create a hauki resource.
+    create_params = {
+        "name": name,
+        "description": description,
+        "address": address,
+        "resource_type": resource_type,
+        "origins": origins,
+        "is_public": is_public,
+        "timezone": timezone,
+        "organization": "tprek:0c71aa86-f76c-466b-b6f3-81143bd9eecc",
+    }
+
+    try:
+        create_response = requests.post(
+            HAUKI_API_URL, json=create_params, headers=authorization_headers
+        )
+        return create_response
+    except requests.exceptions.HTTPError as errh:
+        print("Http Error:", errh)
+    except requests.exceptions.ConnectionError as errc:
+        print("Error Connecting:", errc)
+    except requests.exceptions.Timeout as errt:
+        print("Timeout Error:", errt)
+    except requests.exceptions.RequestException as err:
+        print("OOps: Something Else", err)
+    # If try fails, return 400.
+    return Response("Hauki creation failed.", status=status.HTTP_400_BAD_REQUEST)
+
+
+def partially_update_hauki_resource(url, update_params):
+    """
+    Function for updating hauki resources
+    """
+    authorization_headers = {"Authorization": "APIToken " + API_TOKEN}
+
+    try:
+        update_response = requests.patch(
```

Comment:
Add some small but reasonable timeout value here. We don't want to wait too long here.